File: app/src/main/java/com/zhouyu/android_create/views/RollTextView.java
package com.zhouyu.android_create.views;

import android.annotation.SuppressLint;
import android.content.Context;
import android.util.AttributeSet;
import android.widget.TextView;

/**
 * 自动滚动文字
 */
@SuppressLint("AppCompatCustomView")
public class RollTextView extends TextView {
    public RollTextView (Context context) {
        super(context);
    }
    public RollTextView (Context context, AttributeSet attrs) {
        super(context, attrs);
    }
    public RollTextView (Context context, AttributeSet attrs, int defStyle) {
        super(context, attrs, defStyle);
    }

    @Override
    public boolean isFocused() {
        return isOpen;
    }

    private boolean isOpen = true;

    public boolean isOpen() {
        return isOpen;
    }

    public void setOpen(boolean open) {
        isOpen = open;
    }
}

File: app/src/main/java/com/zhouyu/pet_science/views/RollTextView.kt
package com.zhouyu.pet_science.views

import android.annotation.SuppressLint
import android.content.Context
import android.util.AttributeSet
import android.widget.TextView

/**
 * 自动滚动文字
 */
@SuppressLint("AppCompatCustomView")
class RollTextView : TextView {
    constructor(context: Context?) : super(context)
    constructor(context: Context?, attrs: AttributeSet?) : super(context, attrs)
    constructor(context: Context?, attrs: AttributeSet?, defStyle: Int) : super(
        context,
        attrs,
        defStyle
    )

    override fun isFocused(): Boolean {
        return isOpen
    }

    private var isOpen: Boolean = true
    fun setOpen(isOpen: Boolean) {
        this.isOpen = isOpen
        requestLayout()
    }
}
